test/net/sourceforge/kolmafia/MaximizerTest.java

```diff
@@ -120,7 +120,6 @@ public void freshCharacterShouldNotRecommendEverythingWithCurrentScore()
 
 	// Sample test for https://kolmafia.us/showthread.php?23648&p=151903#post151903.
 	@Test
-	@Ignore("Currently failing")
 	public void noTieCanLeaveSlotsEmpty()
```

Comment:
Because I am still navigating the review process and tools, it is not clear to me whether this test passes with the changes or not.  If it is untested because of Gradle then perhaps ant could be used to run this test against the fixes?  I am not sure that "No tie can leave slots empty" is a correct description of the desired behavior.  Would it be reasonable to want -tie to leave a slot empty if it is currently empty?  If the Surprising Fist skills are perm'd sometimes it makes sense to use them instead of a weapon.

Reply:
I've always felt like -tie should leave slots empty.

Reply:
This particular test now passes, as you can see by the checks below.

Reply:
To better explain the test case naming: it should be parsed as:

"no tie" (i.e. -tie) can leave slots empty

A better name might be "noTiebreakerCan...", since it's implicit that because there is no tiebreaker expression, there is a tie, which should be broken in favor of making no changes.

The intended preference of -tie isn't actually to leave slots empty, but rather leave them unchanged. But in this case, empty == unchanged.